operator/pkg/nodeipam/nodesvclb.go

```diff
@@ -172,6 +186,43 @@ func (r *nodeSvcLBReconciler) getRelevantNodes(ctx context.Context, svc *corev1.
 		return nil, err
 	}
 
+	var nodes corev1.NodeList
+
+	// Special handling for "virtual" Endpoints like Gateways and Ingresses
+	// If the Service has either label that Cilium injects, use all nodes, or filter by Annotation value
+	// https://github.com/cilium/cilium/pull/30038#issuecomment-1970423094
+	managedLabels := sets.New[string](ciliumIngressLabelKey, ciliumGatewayLabelKey)
+	for key := range svc.Labels {
+		if managedLabels.Has(key) {
+			// Look for the presence of the Node filter. If the Annotation is set, parse it to
+			// ensure validity, then use in call to obtain the relevant subset of nodes
+			var lo = client.ListOptions{}
+			if val, ok := svc.Annotations[nodeSvcLBMatchLabelsAnnotation]; ok {
```

Comment:
I think this new annotation should also work in the general path as it seems useful if you want to restrict further nodes. Even with the associated docs that you added not adding this to the general path might still be somewhat confusing for users IMO.

Reply:
I agree. Having this consistent across both Cilium and non-Cilium load balancer services would make the experience better.